tools/matRad_gammaIndex.m

```diff
@@ -0,0 +1,273 @@
+tmpCube = zeros(size(cubex1)); 
+tmpCube((1+searchX):(end-searchX), ...
+        (1+searchY):(end-searchY), ...
+        (1+searchZ):(end-searchZ)) = inf;
+
+
+gammaCubeSq = zeros(size(cubex1));
+gammaCubeSq((1+searchX):(end-searchX), ...
+            (1+searchY):(end-searchY), ...
+            (1+searchZ):(end-searchZ)) = inf;
+
+% find relevant indices for computation...
+ix = cubex1 > 0;
+
+% adjust dose threshold
+if strcmp(localglobal,'local')
+    doseThreshold = cubex1 .* relDoseThreshold/100;                
+elseif strcmp(localglobal,'global')
+    doseThreshold = relDoseThreshold/100 * max(cube1(:));
+end
+
+% search for min
+for i = -searchX:searchX
+    for j = -searchY:searchY
+        for k = -searchZ:searchZ
+            
+            delta_sq = ((i*resolution(1))^2 + ...
+                        (j*resolution(2))^2 + ...
+                        (k*resolution(3))^2) / dist2AgreeMm^2;                 
+            
+            tmpCube((1+searchX):(end-searchX), ...
+                    (1+searchY):(end-searchY), ...
+                    (1+searchZ):(end-searchZ)) = ...
+                             cubex1((1+searchX):(end-searchX), ...
+                                    (1+searchY):(end-searchY), ...
+                                    (1+searchZ):(end-searchZ)) ...
+                           - cubex2((1+((2^n)*searchX)+i) : 2^n : (end-((2^n)*searchX)+i), ...
+                                    (1+((2^n)*searchY)+j) : 2^n : (end-((2^n)*searchY)+j), ...
+                                    (1+((2^n)*searchZ)+k) : 2^n : (end-((2^n)*searchZ)+k));
+                    
+            tmpCube = tmpCube.^2 ./ doseThreshold.^2 + delta_sq;
+            
+            gammaCubeSq(ix) = min(gammaCubeSq(ix),tmpCube(ix));
+            
+            
+        end
+    end
+    
+    display '.';
+    
+end
+
+% evaluate gamma cube and set to zero all the voxel that contain an
+% infinite value
+gammaCubeSqx                 = zeros(size(cube1));
+gammaCubeSqx(cut1,cut2,cut3) = gammaCubeSq((1+searchX):(end-searchX), ...
+                                           (1+searchY):(end-searchY), ...
+                                           (1+searchZ):(end-searchZ));
+gammaCube                    = sqrt(gammaCubeSqx);
+
+% set values where we did not compute gamma keeping inf in the cube to zero
+gammaCube(cube1<=0) = 0;
+
+% need to modify the absDoseThreshold adding a 
+if strcmp(localglobal,'local')
+    doseThresholdX = zeros(size(cube1));
+    doseThresholdX(cut1,cut2,cut3) = doseThreshold((1+searchX):(end-searchX), ...
+                                                   (1+searchY):(end-searchY), ...
+                                                   (1+searchZ):(end-searchZ));
+    doseThreshold = doseThresholdX;
+end
+  
+% compute gamma pass rate
+doseIx          = cube1 > doseThreshold;
+numOfPassGamma  = sum(gammaCube(doseIx) < 1);
+gammaPassRate   = 100 * numOfPassGamma / sum(doseIx(:));
```

Comment:
It would also be cool to calculate gamma pass rates for individual structures (array), since dividing over the number of voxels is often not realy meaningful.